### ISSUE TITLE
Can't FTP a module, but must use Back office 'upload'

### ISSUE THREAD

**Issue by cmroanirgo** [PrestaShop/PrestaShop#10819]
**Describe the bug**
If you FTP (or copy via network, etc) a module into the modules folder, it never shows up in the Back Office, however when you have a .zip file and use the 'upload a module' button, it does.

**To Reproduce**

1. (optional) Use a module created by https://validator.prestashop.com/generator, as I have.
2. Unzip and upload/copy the contents of the zip to the modules folder, checking for correct permissions.
3. Go to 'Modules > Modules & Services'
4. The module is not visible anywhere. Refreshing does nothing.
5. If you delete the folder, and then use the 'upload a module', the sames files are put into the same place. (NB: that access permissions are set to 777 using this method!!!)

(edited by @matks) **Read the discussion below, it explains the issue behind this bug report** 😉 
 
**Additional information**
PrestaShop version:  1.7.4.2
PHP version: 7.x

**Comment by PierreRambaud** [comment]
:+1:  it could be a good improvement. 
@colinegin Already talk about this to Cecile, she agreed :D

**Comment by cmroanirgo** [comment]
I'm not sure if the labels are correct. To me, this is a bug.

Have a look at the developer documentation (https://devdocs.prestashop.com/1.7/modules/creation/)

> At this stage, if you place the module’s folder on the /modules folder, the module can already be seen in the “Modules” page in the back office, in the “Other modules” section – albeit with no real name nor thumbnail.

This clearly isn't happening, and truthfully, being new to Presta development, I've no idea how to realistically develop a module if the only way is to upload a zip file all the time! Nevertheless, the general feel of the documentation is to work in the local filesystem in the modules folder and you'll be able to see the changes in BO. 

I've tried turning off all caching, etc in Performance tab, but to no avail.

It seems I've missed something fundamental...?

**Comment by PrestaEdit** [comment]
A big thing, yes.

It's not the ZIP file you need to have on the /modules folder, as you can read into the doc (the part you mention). It's the content of this ZIP files.

**Comment by Quetzacoalt91** [comment]
Thank you @PrestaEdit, I forgot to mention that.

**Comment by kpodemski** [comment]
I remember that the main issue with that solution was the fact that it was needed to scan entire /modules folder for new modules which was time consuming

**Comment by cmroanirgo** [comment]
@kpodemski According to https://stackoverflow.com/a/41521863 then using `DirectoryIterator` shouldn't be too bad, especially if you have a cached value of the previous time you've scanned and use that as comparison. The tricky bit in that case would be determining what's new if you have no previously cached value, for which you must assume all are candidates, and then use other metrics to decide what is the best course of action.

That said, opening the 'modules' page at the moment is rather slow anyhow, so a few millisecs would disappear easily. (I've noticed the same slowness whether it be local VM or a fast server)

**Comment by hibatallahAouadni** [comment]
@PierreRambaud I think this issue status should be **`To Do`** instead of **`TBS`**, is that so?

**Comment by matks** [comment]
Hi, I just discover this issue (thanks to @hibatallahAouadni) and I am wondering if it's still relevant 🤔 ?

When I want to upload a module directly by folder, what I do is:
1. Move module into `/modules`
2. Open Back-Office page "Improve / Modules / Modules Catalog"
3. I see my module in the list (if needed, use search bar to filter list content)
4. Click "install"

After install has been done, module will be visible in "Improve / Modules / Modules Manager"

I do this when working on DEV mode, I dont know if it is doable in PRODUCTION mode.

**Comment by PierreRambaud** [comment]
> Hi, I just discover this issue (thanks to @hibatallahAouadni) and I am wondering if it's still relevant  ?
> 
> When I want to upload a module directly by folder, what I do is:
> 
> 1. Move module into `/modules`
> 2. Open Back-Office page "Improve / Modules / Modules Catalog"
> 3. I see my module in the list (if needed, use search bar to filter list content)
> 4. Click "install"
> 
> After install has been done, module will be visible in "Improve / Modules / Modules Manager"
> 
> I do this when working on DEV mode, I dont know if it is doable in PRODUCTION mode.

It's also possible in production mode. 
The real problem was a misunderstanding between the catalog and the manager, where the last one is logically also managing your modules present on the disk.

**Comment by matks** [comment]
Maybe we should modify the title of this issue? The title should contain what needs to be done.

Possible solutions:
- display the list of "modules not installed but exist in `/modules`" in Modules Manager in 1st tab
- create a new tab "modules not installed but exist on disk" for these modules in Modules Manager
- add an information message in Modules Manager to tell people "yes, there are modules not installed but exist on disk, you can install them from the Modules Catalog"

then it's "waiting for PM" to tell us the behavior

**Comment by fedesib** [comment]
> Hi, I just discover this issue (thanks to @hibatallahAouadni) and I am wondering if it's still relevant 🤔 ?
> 
> When I want to upload a module directly by folder, what I do is:
> 
>     1. Move module into `/modules`
> 
>     2. Open Back-Office page "Improve / Modules / Modules Catalog"
> 
>     3. I see my module in the list (if needed, use search bar to filter list content)
> 
>     4. Click "install"
> 
> 
> After install has been done, module will be visible in "Improve / Modules / Modules Manager"
> 
> I do this when working on DEV mode, I dont know if it is doable in PRODUCTION mode.

Hi everyone,
I was using this "trick" and it was working perfectly, but in PS 1.7.8.10 theres no there's no search bar in Modules Catalog even if you disable the connection to the Addons (by disabling the module "PrestaShop Marketplace in your Back Office"), so there's no way to "search" for your uploaded module, or at least I didn't find one. 
Any suggestion?

**Comment by fedesib** [comment]
Hi everyone,
I solved my problem by completely uninstalling ps_mbo module -> this gave me back the "old" Modules Catalog page with the search bar and everything and I was able to search uninstalled modules again.

Anyway I think uninstalled modules should appear in the modules list as all the others, it is very handy if you are developing your own module and continuously installing and uninstalling for testing purposes.
Thanks.
